plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/typelibrary/impl/AdapterTypeEntryImpl.java

```diff
@@ -54,12 +55,23 @@ public EClass getTypeEClass() {
 	}
 
 	@Override
-	public synchronized void setType(final LibraryElement type) {
+	protected synchronized NotificationChain basicSetType(final LibraryElement type,
+			final NotificationChain notifications) {
 		if (type instanceof final AdapterType adpType) {
-			// wenn we get a new type ensure that the plug is the mirror of the socket
+			// ensure that the plug is the mirror of the socket
 			adpType.setPlugType(createPlugType(adpType));
 		}
-		super.setType(type);
+		return super.basicSetType(type, notifications);
+	}
+
+	@Override
+	protected synchronized NotificationChain basicSetTypeEditable(final LibraryElement type,
```

Comment:
For the editable type we don't need the plug and as we want to get rid of it I would prefer to not add it anymore.

Reply:
Makes sense, removed.